thread/src/main/kotlin/io/github/paulgriffith/kindling/thread/model/ThreadDump.kt

```diff
@@ -41,7 +40,8 @@ data class ThreadDump internal constructor(
                 }
 
                 ThreadDump(
-                    version = requireNotNull(versionPattern.find(firstLine)?.value),
+                    version = versionPattern.find(firstLine)?.value
+                        ?: throw ToolOpeningException("No version, not a thread dump"),
```

Comment:
Throwing `ToolOpeningException` instead of using `require` here for safer catches downstream